### PR TITLE
completions: improve Blender

### PR DESCRIPTION
**Pull request by PiOverFour**
## Description

Completions for Blender were added in #9905. This commit adds many improvements to these completions:

- Replace short options with old-style options: even though they are single-letter, Blender's options cannot be combined.
- Add comments to separate blocks of options, mirroring Blender's help message.
- Add missing options: render-frame, python-use-system-env, register-allusers, unregister, unregister-allusers.
- Remove options: debug-gpu-disable-ssbo, debug-gpu-renderdoc, -R. Those are no longer available.
- Add animation playback options: -a is a subcommand not compatible with the main ones. Related options use a condition to check for the presence of the -a option.
- Fix typos:
  - frame-end (was end-start)
  - no-window-focus (was no-native-pixels)
  - debug-depsgraph-eval (was -evel)
  - debug-depsgraph-pretty (was -time)
  - app-template (was open-last)
- Update output formats:
  - Add descriptions.
  - Add HDR, TIFF, OpenEXR, OpenEXR Multilayer, FFmpeg, Cineon, DPX, JPEG 2000, and WebP, which are optional but generally available.
  - Remove IRIZ, which is no longer available.
- Fix arguments for --use-extension: they should be 0 or 1, not true or false.
- Make env-system-* options require a parameter.
- Improve --add-ons by querying the list of add-ons inside Blender rather than a hardcoded path. This is because Blender's add-on modules may come from many different paths which depend on platform.
- Fix __blender_echo_input_file_name, by relying on extension.
- Fix listing of scene datablocks inside Blend file.
- Add listing of Python text datablocks to execute inside Blend file.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

**Comment by EmilyGraceSeville7cf** [comment]
Thanks for this improvement, LGTM. 😄

**Comment by PiOverFour** [comment]
> Thanks for this improvement, LGTM. 😄

Thanks for adding the completions in the first place ;)

File: share/completions/blender.fish [hunk 1]
```diff
@@ -1,124 +1,183 @@
...
-    set -l current_token (commandline -t)
-
-    if test "$previous_token" = --addons
-        __blender_list_addons |
-            string replace -r '^' $current_token |
-            string replace -r '$' ','
+    set -l current_token (commandline -t | string replace -r '[^,]*$' '')
+    set -l addons (__blender_list_addons)
+    for a in $addons
+        echo "$current_token$a,"
```
Comment (krobelus) [inline share/completions/blender.fish:45]: prefer to use `__fish_append`

Comment (PiOverFour) [reply share/completions/blender.fish]: Thanks! I didn’t know about it.

**Pull request closed without merging**